test/Streamly/Test/Internal/Data/Fold.hs

```diff
@@ -96,6 +96,9 @@ safeLast (_:xs) = safeLast xs
 last :: [String] -> Expectation
 last ls = S.fold FL.last (S.fromList ls) `shouldReturn` safeLast ls
 
+mapMaybe :: Expectation
+mapMaybe = S.fold (F.mapMaybe (\x -> case x of  5 -> Just x ; _ -> Nothing) F.sum)
+    (S.enumerateFromTo 1 10) `shouldReturn` (5 :: Int)
```

Comment:
We can write this as `mapMaybe :: Int -> Expectation`, then generate a list `xs = [1..n]`. Then we can map `(\x -> if even x then Just x else Nothing)` on `fromList xs` and compare that the result is the same as `filter even xs`.